src/System.Runtime/tests/System/Char.cs

```diff
@@ -5,516 +5,502 @@
 using System;
 using System.Collections.Generic;
 using System.Globalization;
-
 using Xunit;
 
 public static class CharTests
 {
-    [Fact]
-    public static void TestCompareTo()
+    [Theory]
+    [InlineData('h', 'h', 0)]
+    [InlineData('h', 'a', 1)]
+    [InlineData('h', 'z', -1)]
+    [InlineData('h', null, 1)]
+    public static void TestCompareTo(char c, object value, int expected)
     {
-        // Int32 Char.CompareTo(Char)
-        char h = 'h';
-        Assert.True(h.CompareTo('h') == 0);
-        Assert.True(h.CompareTo('a') > 0);
-        Assert.True(h.CompareTo('z') < 0);
+        if (value is char)
+        {
+            Assert.Equal(expected, Math.Sign(c.CompareTo((char)value)));
+        }
+        IComparable comparable = c;
+        Assert.Equal(expected, Math.Sign(comparable.CompareTo(value)));
     }
 
     [Fact]
-    public static void TestSystemIComparableCompareTo()
+    public static void TestCompareTo_Invalid()
     {
-        // Int32 Char.System.IComparable.CompareTo(Object)
-        IComparable h = 'h';
-        Assert.True(h.CompareTo('h') == 0);
-        Assert.True(h.CompareTo('a') > 0);
-        Assert.True(h.CompareTo('z') < 0);
-        Assert.True(h.CompareTo(null) > 0);
+        IComparable comparable = 'h';
+        Assert.Throws<ArgumentException>(null, () => comparable.CompareTo("H")); // Value not a char
+    }
 
-        Assert.Throws<ArgumentException>(() => h.CompareTo("H"));
+    [Fact]
+    public static void TestConvertFromUtf32()
+    {
+        VerifyConvertFromUtf32(0x10000, "\uD800\uDC00");
+        VerifyConvertFromUtf32(0x103FF, "\uD800\uDFFF");
+        VerifyConvertFromUtf32(0xFFFFF, "\uDBBF\uDFFF");
+        VerifyConvertFromUtf32(0x10FC00, "\uDBFF\uDC00");
+        VerifyConvertFromUtf32(0x10FFFF, "\uDBFF\uDFFF");
+        VerifyConvertFromUtf32(0, "\0");
+        VerifyConvertFromUtf32(0x3FF, "\u03FF");
+        VerifyConvertFromUtf32(0xE000, "\uE000");
+        VerifyConvertFromUtf32(0xFFFF, "\uFFFF");
```

Comment:
Why is this not a theory?  Is it due to the xunit issue around serializing string arguments with characters invalid in XML?